pom.xml

```diff
@@ -1771,5 +1772,75 @@
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.jacoco</groupId>
+                        <artifactId>jacoco-maven-plugin</artifactId>
+                        <version>${version.jacoco.plugin}</version>
+                        <executions>
+                            <execution>
+                                <id>ts.jacoco-prepare</id>
+                                <phase>process-test-classes</phase>
+                                <goals><goal>prepare-agent</goal></goals>
+                                <configuration>
+                                    <append>true</append>
+                                    <destFile>target/jacoco.exec</destFile>
+                                    <includes>
+                                        <include>*</include>
+                                    </includes>
+                                    <excludes>
+                                        <exclude>org.jboss.as.test.*</exclude>
+                                    </excludes>
+                                    <output>file</output>
+                                    <propertyName>surefire.jacoco.args</propertyName>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <!-- Ant plugin. -->
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-antrun-plugin</artifactId>
+                        <executions>
+                            <!-- DEBUG -->
+                            <execution>
+                                <id>ts.jacoco.debug</id>
+                                <phase>post-integration-test</phase>
+                                <goals><goal>run</goal></goals>
+                                <inherited>false</inherited>
+                                <configuration>
+                                    <target>
+                                        <echo>Jacoco jvm args: ${surefire.jacoco.args}</echo>
+                                    </target>
+                                </configuration>
+                            </execution>
+                        </executions>
+                        <dependencies>
+                            <dependency>
+                                <groupId>org.jacoco</groupId>
+                                <artifactId>org.jacoco.ant</artifactId>
+                                <version>${version.jacoco.plugin}</version>
+                            </dependency>
+                        </dependencies>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile> <!-- surefire.jacoco.args property needs to be set correctly, if coverage profile is not enabled -->
```

Comment:
Why is a separate profile needed for this, instead of what we do with surefire.jpda.args at L160?

Reply:
Hi Brian.

surefire.jpda.args property is set in jpda profile directly in <properties>
https://github.com/wildfly/wildfly-core/blob/master/pom.xml#L1730

surefire.jacoco.args property is set by JaCoCo plugin:
https://github.com/marekkopecky/wildfly-core/blob/WFCORE-3201/pom.xml#L1804

Although echo from maven-antrun-plugin is able to print this property correctly, JaCoCo plugin is unfortunately unable to set surefire.jacoco.args to the rest of maven with <surefire.jacoco.args/> defined like surefire.jpda.args is defined on L160. The consequence is that surefire JVM and EAP JVM is started without jacoco property and coverage is not stored to exec file.

See this experiment: https://paste.fedoraproject.org/paste/6zSiMYxcu5Y9nj73TYdSEw/raw